Hibernate jsonb support (#574)
GoogleCloudPlatform/pgadapter: samples/java/hibernate/README.md

| timestamptz / timestamp with time zone | LocalDateTime     |
| bytea                                  | byte[]            |
| date                                   | LocalDate         |
| jsonb                                  | Custom Data Type  |

### JSONB

> Hibernate 5.* does not have native support for JSONB. For this, custom data type
> will have to be added. There are modules available that provide support for this.
> For the sample, we have used [vladmihalcea/hibernate-types](https://github.com/vladmihalcea/hibernate-types).

Reviewer: ```suggestion
Hibernate 5.* does not have native support for JSONB. For this, a custom data type
must be added. This sample uses [vladmihalcea/hibernate-types](https://github.com/vladmihalcea/hibernate-types).
```

|------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Hibernate Version      | Version 5.3.20.Final is supported.                                                                                                                                                                                                                                           |
| Postgresql JDBC driver | Version 42.4.3 is supported.                                                                                                                                                                                                                                                 |
| Schema updates         | Cloud Spanner does not support the full PostgreSQL DDL dialect. Automated schema updates using `hibernate` are therefore not supported. It is recommended to set the option `hibernate.hbm2ddl.auto=none` (or `spring.jpa.hibernate.ddl-auto=none` if you are using Spring). |

> | Generated primary keys | Cloud Spanner does not support `sequences`. Auto-increment primary key is not supported. Remove auto increment annotation for primary key columns. The recommended type of primary key is a client side generated `UUID` stored as a string.                                 |

Reviewer: I think we should add pessimistic locking to this table as well (https://docs.jboss.org/hibernate/stable/core.old/reference/en/html/transactions-locking.html). `SELECT ... FOR UPDATE` is currently not supported by Cloud Spanner, and although there are many other backend features that are also not supported, this is one that is used more often by Hibernate users and therefore deserves a specific mention.

Author: Done
